File: internal/auth/auth_user.go
package auth

import (
	"time"

	"github.com/massivebugs/home-feature-server/db/queries"
)

type AuthUser struct {
	Id         uint32
	Name       string
	LoggedInAt time.Time
	CreatedAt  time.Time
}

func NewAuthUserFromQueries(u *queries.User) *AuthUser {
	user := &AuthUser{
		Id:        u.ID,
		Name:      u.Name,
		CreatedAt: u.CreatedAt,
	}

	return user
}

File: web/src/core/models/auth_user.ts
import type { AuthUserDto } from './dto'

export class AuthUser {
  id: number
  name: string
  loggedInAt: Date
  createdAt: Date
  constructor(authUserDto: AuthUserDto) {
    this.id = authUserDto.id
    this.name = authUserDto.name
    this.loggedInAt = new Date(authUserDto.logged_in_at)
    this.createdAt = new Date(authUserDto.created_at)
  }
}
